OMEdit/OMEditGUI/Search/SearchWidget.cpp

```diff
@@ -166,9 +185,26 @@ void SearchWidget::searchInFiles()
   if(mpSearchStringComboBox->currentText().isEmpty() | (mpSearchScopeComboBox->count() == 1)) {
     return;
   }
-  mpSearchStackedWidget->setCurrentIndex(1);
   updateComboBoxSearchStrings(mpSearchStringComboBox);
   updateComboBoxSearchStrings(mpSearchFilePatternComboBox);
+  /* create a new instance of searchresult widget and search widget for every new search */
+  mpSearchWidgetResult = new SearchWidgetResult;
+  pSearchWidgetResultobjects.append(mpSearchWidgetResult);
+  mpSearch = new Search(this);
+  connect(mpSearch, SIGNAL(setTreeWidgetItems(SearchFileDetails)), mpSearchWidgetResult, SLOT(updateTreeWidgetItems(SearchFileDetails)));
+  connect(mpSearch, SIGNAL(setProgressBarRange(int)), mpSearchWidgetResult, SLOT(updateProgressBarRange(int)));
+  connect(mpSearch, SIGNAL(setProgressBarValue(int,int)), mpSearchWidgetResult, SLOT(updateProgressBarValue(int,int)));
+  connect(mpSearch, SIGNAL(setFoundFilesLabel(int)), mpSearchWidgetResult, SLOT(updateFoundFilesLabel(int)));
+  connect(mpSearch, SIGNAL(setProgressBarCancelValue(int,int)), mpSearchWidgetResult, SLOT(updateProgressBarCancelValue(int,int)));
+  connect(mpSearchWidgetResult, SIGNAL(setCancelSearchResult()), mpSearch, SLOT(updateCancelSearch()));
+  /*emit the signals to stop any ongoing search operation when mainwindow is closed*/
+  connect(this, SIGNAL(setCancelSearch()), mpSearch, SLOT(updateCancelSearch()));
+
+  mpSearchStackedWidget->addWidget(mpSearchWidgetResult);
+  mpSearchStackedWidget->setCurrentWidget(mpSearchWidgetResult);
+  QString searchHistoryItem = QString("%1%2%3 %4").arg("Project-").arg(mpSearchScopeComboBox->currentText()).arg(":").arg(mpSearchStringComboBox->currentText());
```

Comment:
Why not,
`QString("%1-%2: %3").arg(tr("Project")).arg(mpSearchScopeComboBox->currentText()).arg(mpSearchStringComboBox->currentText());`